### ISSUE TITLE
Why ngAfterViewChecked is called even ChangeDetectorRef.detach() is fired?

### ISSUE THREAD

**Issue by orange4glace** [angular/angular#10175]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
when ref.detach() is called, until then view is not being updated.
then, in common sense, view is never changing so ngAfterViewChecked also should not be called.
but simply, when setInterval or setTimeout is called, ngAfterViewChecked is also fired.

**Expected/desired behavior**
ngAfterViewChecked should be stop being fired after ref.detach() is called.

**Reproduction of the problem**
https://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5

**What is the expected behavior?**
ngAfterViewChecked should be stop being fired after ref.detach() is called.

**What is the motivation / use case for changing the behavior?**
AmCharts uses very short interval to check user interaction or something.
When I insert this to component, since it calls setInterval every single tick, ngAfterViewChanged is also called every single tick although there's no changes to apply.

**Please tell us about your environment:**
Windows 10, Chrome 51.0.2704.103 m

* **Angular version:** 2.0.0-rc.4
 
* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 
Windows 10, Chrome 51.0.2704.103 m
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5 | Dart] 
Typescript 1.8.10

**Comment by laco0416** [comment]
> Angular version: 2.0.0-rc.4

your plunker repro seems using `2.0.0-beta.9`, too old.
Please try it with the latest version.

**Comment by orange4glace** [comment]
@laco0416 
true, but I'm not good at plunker so I have no idea how to use latest version on it.
And I have tested on my local server with 2.0.0-rc.4 and the problem remains same.

**Comment by laco0416** [comment]
@orange4glace you can use rc.4 with the latest ng2 template :)

![image](https://cloud.githubusercontent.com/assets/1529180/16977962/4b46e9b4-4e94-11e6-994c-8d193ca2fe93.png)

**Comment by orange4glace** [comment]
@laco0416 Wow Thank you so much :)
here is my new one 
https://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5
the problem is still same though

**Comment by vicb** [comment]
See https://github.com/angular/angular/issues/7054 - TLDR this is the expected behavior.

(your last plunker is empty BTW)

**Comment by orange4glace** [comment]
@vicb but If then, every expressions on the template are re-evaluated every time although it doesn't needed. some of expressions also have function like {{timeToString(value)}}. in this case, every time functions are executed, so the performance will goes bad. I have very short period (0 second) interval timer,and I can't remove this.  In this situation, removing function from expressions will ease performance?

**Comment by orange4glace** [comment]
@vicb what I need is this
https://s31.postimg.org/yvcazo2vv/image.png

**Comment by orange4glace** [comment]
In my humble opinion, ref.detach() should remove connection between its parent component. (not entirely) so whether the child has change or not, parent should stay without dirty-check. and when detectChanges() is called, then child and parent should run dirty-checking and update their view.
